recipes/NAS/orion_search.py

```diff
@@ -0,0 +1,564 @@
+
+            losses_m.update(reduced_loss.item(), input.size(0))
+            prec1_m.update(prec1.item(), output.size(0))
+            prec5_m.update(prec5.item(), output.size(0))
+
+            batch_time_m.update(time.time() - end)
+
+            if last_batch or batch_idx % log_interval == 0:
+                log_name = "Test" + log_suffix
+                logging.info(
+                    "{0}: [{1:>4d}/{2}]  "
+                    "Time: {batch_time.val:.3f} ({batch_time.avg:.3f})  "
+                    "Loss: {loss.val:>7.4f} ({loss.avg:>6.4f})  "
+                    "Prec@1: {top1.val:>7.4f} ({top1.avg:>7.4f})  "
+                    "Prec@5: {top5.val:>7.4f} ({top5.avg:>7.4f})".format(
+                        log_name,
+                        batch_idx,
+                        last_idx,
+                        batch_time=batch_time_m,
+                        loss=losses_m,
+                        top1=prec1_m,
+                        top5=prec5_m,
+                    )
+                )
+            end = time.time()
+
+    metrics = OrderedDict(
+        [("loss", losses_m.avg), ("prec1", prec1_m.avg), ("prec5", prec5_m.avg)]
+    )
+    return metrics
+
+
+def objective(alpha, beta, B0, t_zero, res, epochs):
+    batch_size = 32
+    model = PhiNet(
+        input_shape=[3, res, res],
+        alpha=alpha,
+        beta=beta,
+        t_zero=t_zero,
+        num_layers=B0,
+        squeeze_excite=True,
+        h_swish=False,
+        include_top=True,
+        num_classes=nas_args.num_classes,
+    )
+    model.cuda()
+
+    if nas_args.dset == "cifar10":
+        # mean and std of cifar10 dataset
+        TRAIN_MEAN = (0.49139968, 0.48215827, 0.44653124)
+        TRAIN_STD = (0.24703233, 0.24348505, 0.26158768)
+
+    elif nas_args.dset == "cifar100":
+        # mean and std of cifar100 dataset
+        TRAIN_MEAN = (0.5070751592371323, 0.48654887331495095, 0.4409178433670343)
+        TRAIN_STD = (0.2673342858792401, 0.2564384629170883, 0.27615047132568404)
+
+    loader_train, loader_eval = create_loader(
+        image_size=(3, res, res),
+        batch_size=batch_size,
+        data_mean=TRAIN_MEAN,
+        data_std=TRAIN_STD,
+    )
+    logging.info(
+        "Phinet params: %s",
+        {"alpha": alpha, "beta": beta, "t_zero": t_zero, "B0": B0, "res": res},
+    )
+    w = nas_args.w
+    params = torchinfo.summary(
+        model, (1, 3, res, res), device=nas_args.device, verbose=0
+    ).total_params
+    if nas_args.algo == "evo":
+        loss = train_and_evaluate(model, epochs, loader_train, loader_eval)
+        #obj = -1 * loss * (params / nas_args.target) ** w
+        #obj = loss - 100 * np.abs(nas_args.target - params)
+    else:
+        score = score_network(model, batch_size, loader_train)
+        #obj = score * (params / nas_args.target) ** w
+        #obj = score - (np.abs(params - nas_args.target)*1E-6)
+        obj = score*1e-3 - np.abs(params - nas_args.target)*1e-6 
```

Comment:
this is hidden inside the boilerplate, but it's the heart of what we are doing. let's make it a function on its own (see above)